addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -31,7 +31,31 @@ def _l10n_it_edi_generate_electronic_invoice_filename(self, invoice):
            See ES documentation 2.2
         '''
         a = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-        n = invoice.id
+        # Each company should have its own filename sequence. If it does not exist, create it
+        n = self.env['ir.sequence'].with_company(invoice.company_id).next_by_code('l10n.it.edi.fattura.filename')
+        if not n:
+            # start on 1 if the company has no posted invoices (the id may be non-zero due to other invoices from companies)
+            number_next = 1 if self.env['account.move'].search([
+                ('company_id', '=', invoice.company_id.id), ('state', '=', 'posted')
+            ]) == invoice else invoice.id
```

Comment:
This is tricky...  (what if you post 5 invoices together e.g.)  

Should not we try not to find an invoice which is in to_send to fatturapa before this batch somehow...  (not sure either)